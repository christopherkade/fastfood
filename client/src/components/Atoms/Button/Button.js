import React from "react"
import PropTypes from 'prop-types';
import styled from "styled-components"

const StyledButton = styled.button`
  padding: 1rem;
  font-size: 1.5rem;
  background-color: white;
  border: 1px solid #EF798A;
  margin: 0.5rem;
  animation: ${props => props.isShaking ? "shake 0.82s cubic-bezier(.36,.07,.19,.97) both" : "none"};

  & > span {
    font-size: 1.5rem;
  }

  &:hover {
    cursor: pointer;
    opacity: 0.8;
  }

  &:focus {
    outline: transparent;
  }

  @keyframes shake {
    10%, 90% {
      transform: translate3d(-1px, 0, 0);
    }

    20%, 80% {
      transform: translate3d(2px, 0, 0);
    }

    30%, 50%, 70% {
      transform: translate3d(-4px, 0, 0);
    }

    40%, 60% {
      transform: translate3d(4px, 0, 0);
    }
}
`

const Button = ({ children, onClick, isShaking }) => {
  return (
    <StyledButton onClick={onClick} isShaking={isShaking}>
      {children}
    </StyledButton>
  )
}

Button.propTypes = {
  children: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  isShaking: PropTypes.bool
}

export default Button
