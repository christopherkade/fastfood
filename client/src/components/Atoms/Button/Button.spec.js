import React from 'react';
import Button from './Button';
import { render, fireEvent, cleanup } from 'react-testing-library';

afterEach(cleanup)

describe('<Button />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(<Button>Click me</Button>);
      expect(container).toMatchSnapshot();
    });
  });

  describe('children prop', () => {
    it('should display Enter World as Button children', () => {
      const { container } = render(<Button>Click me</Button>);
      expect(container.querySelector('button').textContent).toBe('Click me');
    });
  });

  describe('onClick prop', () => {
    it('should call onClick function when Button is clicked', () => {
      const onClickMock = jest.fn();
      const { getByText } = render(<Button onClick={onClickMock}>Click me</Button>);
      fireEvent.click(getByText('Click me'));
      expect(onClickMock).toHaveBeenCalled();
    });
  });
});
