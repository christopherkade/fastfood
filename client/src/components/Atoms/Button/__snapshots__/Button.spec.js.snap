// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Snapshots should match snapshot 1`] = `
<div>
  <button
    class="sc-bdVaJa anggu"
  >
    Click me
  </button>
</div>
`;
